RIRImageResizing/Classes/NSObject/RIRResizeImageOperation.m

```diff
@@ -0,0 +1,183 @@
+}
+
+#pragma mark - scale
+-(nullable UIImage*)rir_scaleToSize:(CGSize)newSize usingMode:(UIImage_RIRResizing_ResizeMode)resizeMode
+{
+    switch (resizeMode)
+    {
+        case UIImage_RIRResizing_ResizeMode_AspectFit:
+            return [self rir_scaleToFitSize:newSize];
+            break;
+            
+        case UIImage_RIRResizing_ResizeMode_AspectFill:
+            return [self rir_scaleToCoverSize:newSize];
+            break;
+            
+        case UIImage_RIRResizing_ResizeMode_ScaleToFill:
+            return [self rir_scaleToFillSize:newSize];
+            break;
+            
+        default:
+            break;
+    }
+    
+    NSAssert(false, @"unhandled %li",(long)resizeMode);
+    return nil;
+}
+
+-(nonnull UIImage*)rir_scaleToFitSize:(CGSize)newSize
+{
+    /// Keep aspect ratio
+    size_t destWidth, destHeight;
+    if (self.image.size.width > self.image.size.height)
+    {
+        destWidth = (size_t)newSize.width;
+        destHeight = (size_t)(self.image.size.height * newSize.width / self.image.size.width);
+    }
+    else
+    {
+        destHeight = (size_t)newSize.height;
+        destWidth = (size_t)(self.image.size.width * newSize.height / self.image.size.height);
+    }
+    if (destWidth > newSize.width)
+    {
+        destWidth = (size_t)newSize.width;
+        destHeight = (size_t)(self.image.size.height * newSize.width / self.image.size.width);
+    }
+    if (destHeight > newSize.height)
+    {
+        destHeight = (size_t)newSize.height;
+        destWidth = (size_t)(self.image.size.width * newSize.height / self.image.size.height);
+    }
+    
+    return [self rir_scaleToFillSize:(CGSize){
+        .width		= destWidth,
+        .height		= destHeight,
+    }];
+}
+
+-(nonnull UIImage*)rir_scaleToFillSize:(CGSize)newSize
+{
+    UIGraphicsBeginImageContextWithOptions(newSize, NO, self.resizeParameters.scale);
+    
+    [self.image drawInRect:CGRectMake(0,0,newSize.width,newSize.height)];
+    UIImage* newImage = UIGraphicsGetImageFromCurrentImageContext();
+    
+    UIGraphicsEndImageContext();
+    
+    return newImage;
+}
+
+-(nonnull UIImage*)rir_scaleToCoverSize:(CGSize)newSize
+{
+    size_t destWidth, destHeight;
+    CGFloat const widthRatio = newSize.width / self.image.size.width;
+    CGFloat const heightRatio = newSize.height / self.image.size.height;
+    /// Keep aspect ratio
+    if (heightRatio > widthRatio)
+    {
+        destHeight = (size_t)newSize.height;
+        destWidth = (size_t)(self.image.size.width * newSize.height / self.image.size.height);
+    }
+    else
+    {
+        destWidth = (size_t)newSize.width;
+        destHeight = (size_t)(self.image.size.height * newSize.width / self.image.size.width);
+    }
+    
+    return [self rir_scaleToFillSize:CGSizeMake(destWidth, destHeight)];
+}
+
+#pragma mark - resizedImage
+-(UIImage *)resizedImage
+{
+    return [self rir_scaleToSize:self.resizeParameters.newSize usingMode:self.resizeParameters.resizeMode];
```

Comment:
You should make this a read only property, synthesize the instance variable, overload the getter method, and in that method, check if the instance variable is nil, and if it is, generate the resized image and set the instance variable. Let me know if you need to details.

By doing the above, you could call this method more than once, but it only has to generate the image once.